frontend/packages/dev-console/src/components/pipelines/PipelineDetailsPage.tsx

```diff
@@ -13,83 +13,71 @@ import PipelineParameters from './PipelineParameters';
 import PipelineRuns from './PipelineRuns';
 import PipelineForm from './pipeline-form/PipelineForm';
 
-interface PipelineDetailsPageStates {
-  menuActions: Function[];
-  errorCode?: number;
-}
+const PipelineDetailsPage: React.FC<DetailsPageProps> = (props) => {
+  const [menuActions, setMenuActions] = React.useState([]);
+  const [errorCode, setErrorCode] = React.useState(null);
 
-class PipelineDetailsPage extends React.Component<DetailsPageProps, PipelineDetailsPageStates> {
-  constructor(props) {
-    super(props);
-    this.state = { menuActions: [], errorCode: null };
-  }
+  const { name, namespace } = props;
+  const { k8sGet, k8sList } = useSafeK8s();
 
-  componentDidMount() {
-    k8sGet(PipelineModel, this.props.name, this.props.namespace)
+  React.useEffect(() => {
+    k8sGet(PipelineModel, name, namespace)
       .then((res) => {
         // eslint-disable-next-line promise/no-nesting
         k8sList(PipelineRunModel, {
           labelSelector: { 'tekton.dev/pipeline': res.metadata.name },
         })
           .then((listres) => {
-            this.setState({
-              menuActions: [
-                rerunPipeline(
-                  res,
-                  getLatestRun({ data: listres }, 'creationTimestamp'),
-                  'pipelines',
-                ),
-                Kebab.factory.Delete,
-              ],
-            });
+            setMenuActions([
+              rerunPipeline(res, getLatestRun({ data: listres }, 'creationTimestamp'), 'pipelines'),
+              Kebab.factory.Delete,
+            ]);
           })
-          .catch((error) => this.setState({ errorCode: error.response.status }));
+          .catch((error) => setErrorCode(error.response.status));
       })
-      .catch((error) => this.setState({ errorCode: error.response.status }));
-  }
+      .catch((error) => setErrorCode(error.response.status));
+  }, [k8sGet, k8sList, name, namespace]);
```

Comment:
These changes seem to break `PipelineDetailsPage` as well. particularly  `Parameters` and `Resources` tab don't work anymore. This is because of the exhaustive dependency array in the `useEffect`